### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"

### DISCUSSION THREAD

**Discussion by atilavahedian** [community/community#47737] [Copilot Conversations]
### Select Topic Area

Bug

### Body

GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"

I live in iran, I logged out and logged in my account, but it still doesn't work????

**Comment by atilavahedian** [comment]
PLEASE HELP MEEE

**Reply by behroozazarkhalili** [reply]
Have you resolved this issue? If yes, how?
I'm in Linux Ubuntu.
None of vpns worked!

**Reply by M-A-Naderi** [reply]
Hello 
Were you able to solve this problem?

**Reply by RedGuys** [reply]
Only with using VPN

**Reply by mehdikhoram** [reply]
Use this DNS if your are in Iran.

https://403.online/

**Comment by atilavahedian** [comment]
I tested the authorized application and it was accepted, but it still gives the same error.

**Comment by samurainisa** [comment]
The same in Russia

**Reply by korobkin08** [reply]
and in Belarus

**Reply by naruto716** [reply]
Same in New Zealand. I don't think New Zealand is currently under any sanctions.

**Reply by 1stapossum** [reply]
pidory

**Comment by YasinDehfuli** [comment]
This problem is due to sanctions and your country, and preferably try to change your IP.

This may help you : 

https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-common-issues-with-github-copilot

**Reply by naruto716** [reply]
I have the same issue in New Zealand. I don't think New Zealand is currently under any sanctions.

**Reply by golub-kirill** [reply]
The same Winnipeg, MB, Canada. More than 2 month

**Comment by Lippiece** [comment]
Sure, why promote free education, let's go political and make the lives of the people having nothing to do with the events trying to become programmers worse. GitHub -- we have no reason to make you suffer, but we will anyway.

Using VPN doesn't work for me: `GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo EAI_AGAIN api.github.com"`.

**Reply by Thvesh** [reply]
turn on VPN and restart pycharm

**Reply by NeMaksym** [reply]
Your government is literally killing thousands of people. Just, right, now! If you don't support it, go for protests. If you believe it won't help migrate abroad. Otherwise, you just support the terrorist regime with your taxes.
And stop crying about politics. Thinking there is a life outside politics is the reason why your society became such a bloody monster.

**Reply by aalllq** [reply]
Y give money all migrated people ???

**Reply by IWillChangeTheNameLater** [reply]
> Your government is literally killing thousands of people. Just, right, now! If you don't support it, go for protests. If you believe it won't help migrate abroad. Otherwise, you just support the terrorist regime with your taxes. And stop crying about politics. Thinking there is a life outside politics is the reason why your society became such a bloody monster.

Man, what are you talking about? I'm just a student, I have relatives in Ukraine, I pay no taxes, I got access to Copilot with my student license. How can I migrate abroad or go for protests? I'm JUST A STUDENT WHO WANT TO STUDY, that's all.

**Reply by JohnBakhmat** [reply]
> > I pay no taxes
> 
> 
> 
> Do you eat? Because you pay taxes every time you buy smth in a supermarket. You or your parents pay for the dorm and communal services. Your teachers get salary (and pay taxes) because there are students  to teach. You could apply abroad but decided to be part of ru economy.
> 
> Please google how much money someone's life costs the economy.
> 
> 
> 
> > How can I migrate abroad or go for protests? I'm JUST A STUDENT
> 
> 
> 
> Sure. “We just regular people. What can we do against tsar?”. That's the mentality that let USSR exist and that's why ru society become such a failure.
> 
> 
> 
> > How can I migrate abroad or go for protests
> 
> 
> 
> With your feet. Just like millions of internally displaced Ukrainians. You have time to prepare. They hadn't have.

I'm sorry to bother but it's not Twitter so can this politics shit be somewhere else? You have a blank account and come to harass people based on their nationality or citizenship I feel like it's kinda agains ethics or at lease TOS isn't it? ;)

**Reply by NeMaksym** [reply]
> This is unfair and terrible. But that doesn't mean someone else has to suffer.

I said nothing about suffering. It's your statement.
I agree that the “eye for eye” principle doesn't help. But here's a question for you. Imagine a murderer came to your home and killed your family. I assume you're suffering. The murderer is definitely guilty. Should we put him in jail so he feels some “inconvenience” for his actions?

> Why do people have to leave their home because of the regime?

They actually haven't. You can try to make your own country a better place. If you do have not enough power to beat a regime, stop helping it at least. Do what you can. Living a regular life is support. With taxes, your knowledge, your work. Society consists of individuals.

> live in their own country doesn't mean they support the war.

You're right, it doesn't. But again, you're arguing with your own argument.
They may not support a war but contribute to the society that conducts it. And just to remind you, it's not smth hypothetical. Hundreds of people are being killed every day. Just right now.

> how can restricting access to github help

As a separate action, it has minor consequences. But when most companies stop providing modern goods and services to russia, it will first lead to the regression of their society and help some people finally understand "if I want to be part of the developed world, I need to leave this barbaric community". To summarise, it's about resources (human and financial).

**Reply by NeMaksym** [reply]
> You have a blank account

The truthiness of my statements doesn't rely on my account info

 > You ... come to harass people based on their nationality or citizenship
 
 Oh really? Thousands of russian citizens are now on UA territory, killing people every day. It's a fact you can't argue. Saying that out loud is harassment?
I just hope you never experience what the real war is.

> I feel like it's kinda agains ethics or at lease TOS isn't it? ;)

I'm confused. Writing facts about war is against ethics? Well, you may try to report. 
Also, if you care about ethics, I encourage you to donate a few bucks to the funds supporting Ukraine. It may save someone's life.

**Reply by SergeevSergey99** [reply]
> Your government is literally killing thousands of people. Just, right, now! If you don't support it, go for protests. If you believe it won't help migrate abroad. Otherwise, you just support the terrorist regime with your taxes. And stop crying about politics. Thinking there is a life outside politics is the reason why your society became such a bloody monster.

Hey, are you serious? Do you think we don't know something about our government? What country are you from? Do you really think that everything works so easily in dictatorship-like countries? 

Do you think we didn't protest? Have you ever been beaten with batons and tasers, put in a medieval-like prison for weeks because you literally held a poster on which nothing is written? Or maybe you were delivering transfers to those who were captured after the protests because they are almost not fed there? And more famous people are actively being killed. All sources of information have been captured by propaganda. Our social networks are actively monitored by special services and put in jail even for likes. 

Do you think we can emigrate so easily? The majority of the population hardly has enough salaries to live on. The economy is specially built so that almost all the money goes to an apartment and food and you have to work at several jobs. So that there was no time to search for the truth and reflection, and the punishment for the protest affected the whole family as hard as possible. And even if someone can save money for emigration, now because of some stupid sanctions it is almost impossible. We are not given any visas now. Our cards do not work abroad and we cannot withdraw money from the Russian economy. The only way was to get a job abroad, but most international companies have left Russia and no longer hire Russians. But these sanctions will not affect those who do not want to emigrate. Those who believe that the rest of the developed world is the enemy, but why did they have to buy the cheapest products even before the sanctions. And yes, thanks to them, we also can't donate for supporting Ukraine.

Our society is not a bloody monster. It is intimidated, beaten and locked in a cage with a terrorist. Before our government attacked a neighboring country, it took over our country.

**Comment by NeluQi** [comment]
How disappointed I am in you. Fuck your politics

Just because my IP is wrong, they sent me to the ass. The development of technology, science? No, why? We have a policy. What am I to blame? I just want to do IT

**Comment by himark-dev** [comment]
This must be just a bug. Accroding to github policies there are no currently active restrictions for russian students and their student packs including copilot.
https://docs.github.com/ru/site-policy/other-site-policies/github-and-trade-controls

**Reply by Lippiece** [reply]
Well if I try to use it normally, it explicitly says:
```
[INFO] [auth] [2023-02-17T10:02:54.420Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2023-02-17T10:02:54.422Z] GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"
```

**Reply by himark-dev** [reply]
i'm pretty sure this issue will dissapear after a while like all copilot network issues do
since i am facing the same problem I will wait until tomorrow in the hope that the problem will be solved

**Reply by atilavahedian** [reply]
Yes, it is exactly the same for me in Iran

**Comment by atilavahedian** [comment]
Thank you for your comments and help.

**Reply by behroozazarkhalili** [reply]
Have you resolved this issue?

**Reply by RedGuys** [reply]
> Have you resolved this issue?

Only with VPN

**Reply by behroozazarkhalili** [reply]
Which VPN, as I cannot handle it with a VPN proxy either!

**Reply by RedGuys** [reply]
> Which VPN, as I cannot handle it with a VPN proxy either!

I use WireGuard with a rented server in the Netherlands
P.S. Don't forget to restart the IDE after enabling the VPN

**Reply by behroozazarkhalili** [reply]
I use it with a server in Europe as well.
Restart the vscode and the whole system; however, it does not work!

**Reply by eclipse130300** [reply]
Did not work even with VPN. BUT! After switching from proxy to tunnel VPN mode copilot seems to work properly

**Reply by MahdiTa97** [reply]
I had the same issue, but it was resolved by setting my HTTP/socks proxy in VS Code's proxy settings.

**Comment by teuchezh** [comment]
The same problem, just yesterday everything worked without problems, and just today I received an email that access has been extended until February 2024. And I'm also from Russia. But as far as I read in the certificate, sanctions restrictions apply only to Corporate Services and, moreover, not to the student package.

**Comment by dotterian** [comment]
This is what I get for paying yearly subscription with my US card, while residing in Russia.
I wish there were an option to get a refund.

**Comment by ArtemySazonov** [comment]
Same situation: a student located in Moscow, Russia. Let us hope this to be a bug

**Comment by nubovik01** [comment]
pizdec

**Reply by nkarim496** [reply]
nahui blyat

**Comment by Andrew-71** [comment]
Russian here, having issues like others too! D:

Hopefully should disappear on its own, after re-reading policy doesn't seem like there are any restrictions on co-pilot. Otherwise would be really disappointing to see politically motivated feature cuts.

**Reply by BlackMighty** [reply]
> > политически мотивированные сокращения функций?
> 
> Ваше правительство и граждане буквально убивают тысячи людей. Прямо сейчас! А вы живете в россии как ни в чем не бывало, пьете смузи и платите налоги своему террористическому режиму. Думать, что есть жизнь вне политики, вот почему ваше общество превратилось в такого кровавого монстра. Если вам это не нравится, протестуйте или эмигрируйте. В противном случае вы являетесь частью этого.
Братишка что тебе не нравится? Ведь твое правильство тоже убивает людей, вместо того чтобы договариваться. Но почему то лично ты, тут в комментах пишешь, вместо того, чтобы выйти на проест.

**Comment by brolnickij** [comment]
i have same problem

P.S. using VPN makes the problem go away

**Reply by atilavahedian** [reply]
What VPN do you have?

**Reply by brolnickij** [reply]
> What VPN do you have?

ProtonVPN (free Japan servers)

**Reply by atilavahedian** [reply]
Proton not work in iran.

**Reply by brolnickij** [reply]
> Proton not work in iran.

im from Russia :)

**Reply by JohnBakhmat** [reply]
@atilavahedian try tunnelbear

**Reply by Lippiece** [reply]
Confirming Proton VPN Japan works.

**Comment by favelanky** [comment]
The same. I'm from Russia

**Comment by Inzilkin** [comment]
We probably hope it was a bug, not a sanction from Microsoft 👎🏻

**Comment by aliaminibagh** [comment]
**GitHub gave me the following answer:**

Hello Ali,

Thanks for writing us today, and sorry for the difficulty you're experiencing. I'm afraid that due to current trade controls, your GitHub account cannot make use of Copilot.

While GitHub obtained a license in 2021 to provide cloud services to Iranian users, Copilot is not included in the scope of this license.

I'm sorry if that wasn't what you were hoping to hear.

We want every developer to be able to collaborate on GitHub, and are undertaking efforts to increase Copilot availability around the world, so please know that GitHub will continue to advocate for the right of developers in trade-restricted regions to access tools for code collaboration and communication.

Regards,
GitHub Support

**Reply by Armanasq** [reply]
It's ridiculous. I am using Ubuntu and there is no free VPN for it which works in this situation. I have tried Clash and QV2ray, they connect to the server but I still get the same error.

F#ck Politics and Politician

**Reply by fdciabdul** [reply]
use vpnbook.com its free ..

**Reply by Armanasq** [reply]
> use vpnbook.com its free ..

VPNBook has been blocked in Iran. None of its servers works in Iran

**Reply by fdciabdul** [reply]
> > use vpnbook.com its free ..
> 
> VPNBook has been blocked in Iran. None of its servers works in Iran

mmm why .. idk exactly what happen in iran , maybe u can use protonvpn but yea , still u must pay for it

**Comment by standaloneee** [comment]
just download openvpn usa config and relog into copilot plugin in whatever IDE you have, for some period of time it might fix the problem

**Comment by dfg-98** [comment]
Same issue here, I'm from Cuba.

According to [Github and Trade Controls](https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#can-you-clarify-availability-of-github-to-cuban-developers)

> ## Can you clarify availability of GitHub to Cuban developers?
> GitHub cloud services, both free and paid, are generally available to developers located in Cuba.
Specially Designated Nationals (SDNs), other denied or blocked parties under U.S. and other applicable law, and certain government officials may be restricted from accessing or using GitHub, wherever located. Additionally, users may not use GitHub.com for or on behalf of such parties, generally including the Governments of sanctioned countries.

**Comment by stepanskur** [comment]
Вот и гитхаб прогнулся! А так надеялся, что хоть они дадут надежду на нормальную работу...

**Reply by JohnBakhmat** [reply]
> > Вот и гитхаб прогнулся! А так надеялся, что хоть они дадут надежду на нормальную работу...
> 
> 
> 
> Ты серьезно? Без копилота жизнь окончена типо или что? На работу без него не устроишься?)

При чем тут "жизнь окончена"? Вопрос стоит про нормальный DX а не про невозможность кодить.

**Comment by M4hbod** [comment]
Hi there
I just found a solution for Iranian users.
I tried many different ways and using [Nekoray](https://github.com/MatsuriDayo/nekoray) and a server from this [channel](https://t.me/+42AVX1ZnlRQxMmQ5) fixed the problem. You must use VPN Mode instead of System Proxy in Nekoray.
I guess you can also use System Proxy + Proxifier.

**Reply by behroozazarkhalili** [reply]
Let's try mahbod!

**Comment by TungusSs** [comment]
Same problem in Belarus, I guess this is not a "bug" anymore but restrictions by GitHub. It's a shame for GitHub - they cut off students from a normal developer world

**Comment by JohnBakhmat** [comment]
First they use open source code to train the model, then you must pay for it, even if you pay you are not allowed to use the tool if you are not in a "chosen" country.
Good job Microsoft! 👍🏻

**Comment by cinaaaa** [comment]
Stupid sanctions!!! 👎 👎 Microsoft
https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#under-the-license-github-has-received-from-ofac-which-types-of-accounts-will-be-available-in-iran

## Solution
Set an HTTP proxy in the VSCODE preferences that are directed from V2Ray or any other VPN channel.

![Screenshot 2023-02-21 at 12 35 31](https://user-images.githubusercontent.com/42252572/220298972-cfe96442-f8c8-4f5a-88ea-3919402147ef.png)

**Reply by behroozazarkhalili** [reply]
Yeah. It works perfectly!

**Reply by Armanasq** [reply]
> Stupid sanctions!!! -1 -1 Microsoft https://docs.github.com/en/site-policy/other-site-policies/github-and-trade-controls#under-the-license-github-has-received-from-ofac-which-types-of-accounts-will-be-available-in-iran
> 
> ## Solution
> Set an HTTP proxy in the VSCODE preferences that are directed from V2Ray or any other VPN channel.
> 
> ![Screenshot 2023-02-21 at 12 35 31](https://user-images.githubusercontent.com/42252572/220298972-cfe96442-f8c8-4f5a-88ea-3919402147ef.png)

Thanks, It worked for me by using QV2ray + adding proxy to the VScode setting.

**Reply by M-A-Naderi** [reply]
Can you send me the complete solution?

**Comment by fdciabdul** [comment]
use Tabnine , its free

**Reply by Sanikookalam** [reply]
If you use it once , you regret from what you said

**Comment by bbkgh** [comment]
I have same problem

**Comment by tuves** [comment]
We became aware that Copilot was unavailable in several countries following our efforts to restore access to Ukraine, and on further investigation, we identified the cause as originating from a change we deployed to block Copilot in trade-restricted countries. To prevent this from happening again, we've updated our reporting dashboards to better account for error rates due to trade restrictions and are implementing more robust testing of the trade restriction logic.

**Reply by wtf403** [reply]
So, go ahead, Microsoft. Why is github still available in countries where trade is restricted? You can also release an update for Windows that will burn down all computers there. It will be so Microsoftish 😘

**Reply by TungusSs** [reply]
You haven't answered the question. When will Copilot become available again?

**Reply by behroozazarkhalili** [reply]
Are you seriously kidding us?
You first destroyed everything, then tried to find a problem and resolve it?

**Comment by mahdjalili** [comment]
You must use VPN due to sanctions. You can use any VPN you want. But if you are using proxy servers or v2ray fallow these steps:

The problem is that vscode does not use the proxy settings of v2rayN or any other proxy server you have, and the traffic of vscode does not pass through the proxy. you have to set this manually.

Go to settings > application > proxy
Set the proxy server information in Http:Proxy to:
`http://proxyserver:port`
The default proxy server information in v2rayN should be as follows:
`http://127.0.0.1:10809`
   
If the proxy server you want to use requires authentication, set the following commands to configure the proxy server:
`http://username:password@proxyserver:port`

Other Solutions:
- use Nekoray in VPN mode.
- use Proxifier and add vscode to its rules.

**Reply by neginazadi-tech** [reply]
hi, for me this works but I have to pay for its activation. it's free for you?

**Reply by mahdjalili** [reply]
> hi, for me this works but I have to pay for its activation. it's free for you?

If you mean is copilot free? No. I have github pro and that's why I can use copilot. You have to pay a monthly fee to use it. But copilot is free to use for verified students, teachers, and maintainers of popular open source projects.

If you mean is v2ray free? Yes, v2ray is an open source project. But there are people who sell servers or subscriptions for God's sake :)

**Reply by M-A-Naderi** [reply]
Very very thank you its worked for me 💚💚💚

**Reply by M-A-Naderi** [reply]
If you are in Iran, you can extract dns from the site shecan.ir and use it and use it without VPN.

**Reply by hessamkhoobkar** [reply]
Thank you, @mahdjalili. For me, this was the answer. 

When using vmess, copilot did not work for me, while with other protocols, it did. But I could not use other protocols as Firebase gave me a 403 error when I attempted to use them. Consequently, I have no choice but to utilize vmess.

In my case, setting vscode's proxy to `http://127.0.0.1:10809` and unchecking Proxy Strict SSL was enough.
The vmess is now recognized by vscode, so now I can use it.

**Reply by ah-rahimi** [reply]
Thanks For useful help

**Reply by ning2510** [reply]
thx

**Reply by woreom** [reply]
> Thank you, @mahdjalili. For me, this was the answer.
> 
> When using vmess, copilot did not work for me, while with other protocols, it did. But I could not use other protocols as Firebase gave me a 403 error when I attempted to use them. Consequently, I have no choice but to utilize vmess.
> 
> In my case, setting vscode's proxy to `http://127.0.0.1:10809` and unchecking Proxy Strict SSL was enough. The vmess is now recognized by vscode, so now I can use it.

Thank you, @hessamkhoobkar and @mahdjalili. We can skip the Proxifier step by unchecking 'Proxy Strict SSL' in VSCode. Additionally, if you want to know the port for Nekoray, it is displayed under the app's window as HTTP:127.0.0.1:&lt;nekoray-port&gt;.

**Comment by jiftoo** [comment]
VPNs are a blessing in these times 😔

**Comment by Sanikookalam** [comment]
Iran Devs,
Read this ...
https://github.blog/2021-01-05-advancing-developer-freedom-github-is-fully-available-in-iran/

**Comment by AleksandrKrasovski** [comment]
Hi everyone.
This is not politics, this is business. Need to pay for Copilot.
https://github.com/github-copilot/signup

**Reply by RedGuys** [reply]
I have a subscription to copilot, I can't use it without vpn, isn't that a politics?

**Reply by jadehapp** [reply]
Im from Iran, I have the subsciption and paid for Copilot but just cannot use the vscode extention due that error

**Comment by 03timothyash** [comment]
En mi opinión en países se paga porque la información corra limpia como el agua. Y no nada más llege ala mitad esto va a ser bien fuerte pero fluido bueno

Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by BlackMighty** [comment]
What exactly?

  
>Среда, 29 марта 2023, 11:52 +03:00 от Negin Azadi ***@***.***>:
> 
> 
>hi, for me this works but I have to pay for its activation. it's free for you?
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you commented. Message ID:  <community/community/repo-discussions/47737/comments/5463157 @ github . com> 
 
 
--
Андрей Золотарев
Отправлено из Почты  Mail.ru

**Comment by kuklofon** [comment]
Why i can't use copilot?! Today i had this - At this time, Copilot is not available in your location
WTF? VPN doesn't help anymore

**Reply by scyACE** [reply]
me too

**Reply by levinit** [reply]
bro, check the billing page, maybe modify your address or first name , it will be solved, I used a fake address and first name before.

**Reply by kuklofon** [reply]
thkx, i'll try

**Comment by levinit** [comment]
I can not use copilot in goland today, but works
 in vscode and webstorm
, already paid 100 , not free plan... I can not understand why China is an "unavailable location"!? 

---
solved.  should modify profile, do not use a fake address or first name, if not, it may think that you are from some restricted area, the specific reason is not clear, I provide relatively correct name and city name.

**Reply by NoobMidC** [reply]
me too, same in China;

**Reply by scyACE** [reply]
i was using last day,but it not works today in China

**Reply by levinit** [reply]
> me too, same in China; 

我发现原因了，我以前地址写的假的，到billing界面有提示验证不通过，我估计地址验证不通过统一按限制地区处理，改了相对真实的名字和地址就解除限制了。  说明ms开始扫用户信息，控制开始严格起来了，说不定后面也会限制中国，至于r国这种被限制怎么也不行了。

**Reply by levinit** [reply]
> i was using last day,but it not works today in China

看我上一条回复，解决了

**Reply by scyACE** [reply]
> > i was using last day,but it not works today in China
> 
> 看我上一条回复，解决了

牛掰 兄弟 留个邮箱加个联系方式！

**Reply by IgreatD** [reply]
啵啵，牛的

**Reply by kuklofon** [reply]
At this time, Copilot is not available in your location - in Russia

**Reply by NoobMidC** [reply]
> > me too, same in China;
> 
> 我发现原因了，我以前地址写的假的，到billing界面有提示验证不通过，我估计地址验证不通过统一按限制地区处理，改了相对真实的名字和地址就解除限制了。 说明ms开始扫用户信息，控制开始严格起来了，说不定后面也会限制中国，至于r国这种被限制怎么也不行了。

牛的，我改了就好了；哎，希望ms别禁cn吧，神仙打架凡人遭殃很痛苦；

**Reply by levinit** [reply]
> At this time, Copilot is not available in your location - in Russia

I am not clear whether Russia is in "unavailable location" list, can not find any pages tells what is unavailable location, try to set the correct country & city in the payment information page.

**Reply by zhangle9012** [reply]
the same error for student user in China, where can I modify my profile? 
the copilot is free for student user so I don't have any billing page ...

**Reply by levinit** [reply]
> the same error for student user in China, where can I modify my profile? 
> 
> the copilot is free for student user so I don't have any billing page ...

免费的我不清楚，你看看个人设置里面那个billing 项下的payment info，或者你看看你那些个人信息中有关地址和姓名的部分，尝试改成真实的城市名字

**Reply by sirenfrappe** [reply]
我也是蹭的学生的，改了地址之后就能用了

**Reply by zhangle9012** [reply]
我的setting--billing and plans--payment information下面倒是显示了地址，但是似乎没有看到修改的入口？是因为没有添加支付方式吗。。。

**Comment by blkst8** [comment]
I live in Iran and I can't use it anymore even with VPN since last night.
THIS IS THEFT. YOU GET MONEY AND THEN APPLY NEW RULES AGAINST SOME COUNTRIES.

**Reply by levinit** [reply]
> I live in Iran and I can't use it anymore even with VPN since last night.
> 
> THIS IS THEFT. YOU GET MONEY AND THEN APPLY NEW RULES AGAINST SOME COUNTRIES.

hi, see my rely of this issue, modify your payment information on billing page, it will be work perhaps....

**Reply by blkst8** [reply]
Hi and thanks for your help. it works.
thanksssssss. :_____)

**Comment by Jaimeanderson** [comment]
The message "GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"" suggests that the user is attempting to use the Copilot extension from a location where it is currently not available. This may be due to regional restrictions or the extension is not yet available in that particular country.
While this can be frustrating for users who were looking forward to using the extension, it's important to respect regional laws and regulations. It's also possible that GitHub may be working on expanding the availability of the extension in the future.
In conclusion, while it can be disappointing to not have access to certain tools or extensions, it's important to understand and respect regional laws and regulations. Companies like [Casinofindr]([https://casinofindr.com/best-online-casino](https://www.healthsoothe.com/)) prioritize compliance and regulation to ensure a safe and enjoyable experience for all their users.

**Reply by levinit** [reply]
> The message "GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"" suggests that the user is attempting to use the Copilot extension from a location where it is currently not available. This may be due to regional restrictions or the extension is not yet available in that particular country.
> 
> While this can be frustrating for users who were looking forward to using the extension, it's important to respect regional laws and regulations. It's also possible that GitHub may be working on expanding the availability of the extension in the future.
> 
> In conclusion, while it can be disappointing to not have access to certain tools or extensions, it's important to understand and respect regional laws and regulations. Companies like [Casinofindr]([https://casinofindr.com/best-online-casino](https://www.healthsoothe.com/)) prioritize compliance and regulation to ensure a safe and enjoyable experience for all their users.

yes ，you are right, but I can not find any notice or QA about this question, (maybe just I can not find this page)

**Comment by sthetz** [comment]
Regardless of politics, restricting access by IP is just plain retarded because VPNs do exist. Now I have to put up with ~50ms latency to use copilot from my current physical location. Why inconvenience your clients? Isn't restricting by payment method enough for sanctions compliance? 

Update: I am now using VSCode's feature that allows putting all of its traffic through a proxy, including extensions. Now I can use Copilot without having to route all my networking via VPN and suffer the lag. It's a minor inconvenience, but still...

**Comment by jadehapp** [comment]
Im from Iran too

SOLUTION

using VPN in Proxy mode doesn't work because vscode uses your network without it.
use VPN mode if you can. 
if you are using v2ray vpns there is an Application named "NekoRay" that has a VPN mode button, after activate that reopen your vscode and copilot extension then it works.

**Comment by rezzaem** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> GitHub Copilot could not connect to server. Extension activation failed: "At this time, Copilot is not available in your location"
> 
> I live in iran, I logged out and logged in my account, but it still doesn't work????


hello there 🙂
I found the solution !
U can use Shecan dns to slove the "not available in your location" problem.
just visit [www.shecan.ir](https://shecan.ir/) and set the dns on your system

 good luck 🔥💡

**Reply by stepanskur** [reply]
Wow! It works in Russia!!!

**Reply by JohnBakhmat** [reply]
I confirm. It works. Steps:
1. Open the site. In green box theres DNS, DNS over TLS and, DNS over HTTP.
2. Open your (windows) "Network & Internet" > Wi-Fi (or the thing you have).
3. Press "Edit" near "DNS server assignment".
4. Edit IPV4.
5. Write first DNS ip and second as alternative. 
6. I also selected "DNS over HTTP (manual)" option and entered what they gave in both field, BUT i dont know if its essential.

**Comment by JohnBakhmat** [comment]
It's not an outdated thread. You just don't do anything about it. ;)

**Comment by Rryowa** [comment]
**Settings to route all VSCode traffic through your V2rayN client as for 06.2025**
![image](https://github.com/user-attachments/assets/b9796610-2b59-4b42-bc30-edd90a0fb136)
![image](https://github.com/user-attachments/assets/0e8f6402-f617-4733-b615-27ab3206f26b)
